### PR TITLE
Fix #61 : Smoke test / Probot v11 migration key points / Common errors

### LINKED ISSUE #61: README improvments proposal
Following [#58](https://github.com/probot/adapter-aws-lambda-serverless/issues/58), README content is nice for a Probot v11 from scratch.

Here a draft (I will do a PR after acceptance/remarks) for (potentials) README improvments on 3 themes:
1. Synthesis of server-less adapter changes for Probot prior v11 migration (it could be quite easy to forget an item when you are migrating from Probot <=v10)
1. Lambda endpoint URL
1. Smoke test with Probot v11 (Pain point for to propose a PR)

---

//1. This new part between 'Examples' and 'LICENSE'

## Probot v11 migration key points

For Probot v11 support, this adapter introduces significant changes. Here the key points to update (in addition of [Probot v11 breaking changes](https://github.com/probot/probot/releases/tag/v11.0.0)):

| Key point / Probot | <= v10 | >= v11 |
| --- | --- | --- |
| NPM package name | `@probot/serverless-lambda` | `@probot/adapter-aws-lambda-serverless` |
| *handler.js* content | See [Usage v1.x](https://github.com/probot/adapter-aws-lambda-serverless/tree/v1.0.2#usage) | See [Usage](#usage) |
| AWS Lambda Runtime | `handler.probot` | `handler.webhooks` |
| AWS Lambda Handler | Node.js 12.x *(preferred)* | Node.js 12.x *(required)* | 

---

//2. Lambda endpoint URL

> Make sure to configure your GitHub App registration's webhook URL to <your lambda's URL>/api/github/webhooks

The `/api/github/webhooks` is quite disturbing ... from Probot v10 usage it is like we need to add this URL suffix (but no).

I propose to remove it and explain that the *lambda's URL* is the **invoke URL** for POST verb, which can be found in the *Stages* configuration from the *Amazon API Gateway* console.

---

//3. Smoke test with Probot v11

This adapter is amazing, the best way to deploy stateless Probot 😗.

But with the matrix `AwsNodeRuntime + Probot version + Serverless adapter version`, it is sometime hard to troubleshoot a problem, and end2end configuration with a GitHub is not the simplest.

Having the way todo a *smoke test* could be nice. With Probot v10, this one worked fine:

```shell
$ curl --request POST --header "X-Github-Event: test" --data "{\"action\": \"test\"}"  https://x.execute-api.y.amazonaws.com/github-probots/sample-probot
{"message":"Received test.test"}
```

With Probot v11, we should deal with `X-Hub-Signature` header 😢 (Thanks to [Securing your webhooks](https://docs.github.com/en/developers/webhooks-and-events/securing-your-webhooks) ^^).

I would like provide a snippet like that, but I haven't succeeded at this time (is it achievable with `openssl dgst` ?) ... if anybody has some advices, many thanks ! 

```shell
$ LAMBDA_URL=https://x.execute-api.y.amazonaws.com/github-probots/sample-probot
$ TMP_DATA_FILE=/tmp/smoke.data
$
$ echo "{\"action\": \"test\"}" > $TMP_DATA_FILE
$ SIGN=$(openssl dgst -sha1 -hmac $HERE_THE_WEBHOOK_SECRET_??_OR_NOT $TMP_DATA_FILE | cut -d" " -f2)
$ curl --request POST --header "X-Hub-Signature: sha1=$SIGN" --header "X-Github-Event: test" --data-binary "@$TMP_DATA_FILE" $LAMBDA_URL
{"message":"Received test.test"}
```

### PR DESCRIPTION
**Pull request by axel3rd**
-----
[View rendered README.md](https://github.com/axel3rd/adapter-aws-lambda-serverless/blob/doc-improvments/README.md)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 node_modules/
+/.settings/
+/.project
```
Comment (gr2m) [inline .gitignore:3]:
```suggestion
.settings/
.project
```

File: README.md [hunk 2]
```diff
@@ -49,12 +49,45 @@ functions:
...
 - [example-aws-lambda-serverless](https://github.com/probot/example-aws-lambda-serverless/#readme) - Official example application that is continuously deployed to AWS Lambda
 
 Add yours!
 
+## Common errors
+
+| Http Code | Message | Description
+| --- | --- | --- |
+| 403 | Missing Authentication Token | Bad endpoint (this one is not binded on Lambda) |
+| 500 | Internal server error | Incorrect headers value (`X-GitHub-Delivery`, `X-GitHub-Event`, `X-Hub-Signature`) or Probot error |
```
Comment (gr2m) [inline README.md:78]: can you run `npm run lint:fix`? That will format the markdown tables

**Review by gr2m — COMMENTED** [review]
Looks good to me, just a few nits

File: README.md (hunk 2 above)
Comment (axel3rd) [reply README.md]: Done

**Review by gr2m — APPROVED** [review]
Thank you!

**Pull request merged**